service/src/main/java/uk/gov/hmcts/reform/fpl/validators/HasDocumentStatusValidator.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import org.apache.commons.lang.StringUtils;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasDocumentStatus;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class HasDocumentStatusValidator implements ConstraintValidator<HasDocumentStatus, CaseData> {
+    @Override
+    public void initialize(HasDocumentStatus constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(CaseData caseData, ConstraintValidatorContext constraintValidatorContext) {
+        constraintValidatorContext.buildConstraintViolationWithTemplate(constraintValidatorContext
+            .getDefaultConstraintMessageTemplate()).addPropertyNode("documents").addConstraintViolation();
+
+        if (caseData.checklistDocument() != null
```

Comment:
That logic does not look right. If one document is provided all other documents (not only statuses) are required.

The following scenario fails:


```
CaseData caseData = CaseData.builder()
            .documents_socialWorkAssessment_document(Document.builder()
                .documentStatus("Status")
                .build())
            .build();

        List<String> errorMessages = validator.validate(caseData).stream()
            .map(error -> error.getMessage())
            .collect(Collectors.toList());

        assertThat(errorMessages).doesNotContain(ERROR_MESSAGE);
```

due to:

```
not to contain
 <["Tell us the status of all documents including those that you haven't uploaded"]>
but found
 <["Tell us the status of all documents including those that you haven't uploaded"]>
```

Reply:
Is that not to be expected? In documents you must have all documents entered (expect additional) including the status populated. I would expect the error message to appear in the above because all documents have not been populated.

Reply:
Hmm, not sure. It seems looks wired to me. I would double check it with business.

Does it mean we should always retire them to set status of all documents?

Reply:
I've queried this with Theo, this is the case unfortunately. Content could be more specific as to what document is missing etc.